File: app/src/main/java/com/worldventures/dreamtrips/modules/trips/command/CheckTripsByUidCommand.java
package com.worldventures.dreamtrips.modules.trips.command;

import com.worldventures.dreamtrips.core.janet.dagger.InjectableAction;
import com.worldventures.dreamtrips.core.repository.SnappyRepository;

import java.util.List;

import javax.inject.Inject;

import io.techery.janet.Command;
import io.techery.janet.command.annotations.CommandAction;

@CommandAction
public class CheckTripsByUidCommand extends Command<Boolean> implements InjectableAction {

   @Inject SnappyRepository snappyRepository;

   private List<String> uids;

   public CheckTripsByUidCommand(List<String> uids) {
      this.uids = uids;
   }

   @Override
   protected void run(CommandCallback<Boolean> callback) throws Throwable {
      callback.onSuccess(snappyRepository.hasTripsDetailsForUids(uids));
   }
}

File: app/src/main/java/com/worldventures/dreamtrips/modules/trips/service/command/CheckTripsByUidCommand.kt
package com.worldventures.dreamtrips.modules.trips.service.command

import com.worldventures.dreamtrips.core.repository.SnappyRepository
import com.worldventures.janet.injection.InjectableAction
import io.techery.janet.Command
import io.techery.janet.command.annotations.CommandAction
import javax.inject.Inject

@CommandAction
class CheckTripsByUidCommand(private val uids: List<String>) : Command<Boolean>(), InjectableAction {

   @Inject internal lateinit var snappyRepository: SnappyRepository

   @Throws(Throwable::class)
   override fun run(callback: Command.CommandCallback<Boolean>) {
      callback.onSuccess(snappyRepository.hasTripsDetailsForUids(uids))
   }
}
